server/src/main/kotlin/suwayomi/tachidesk/server/ServerConfigDb.kt

```diff
@@ -0,0 +1,80 @@
+package suwayomi.tachidesk.server
+
+/*
+ * Copyright (C) Contributors to the Suwayomi project
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import kotlin.reflect.KProperty
+import kotlin.reflect.KProperty1
+
+abstract class DatabaseDelegatorModule {
+    protected var databaseDelegator = DatabaseDelegator()
+}
+
+class DatabaseDelegator {
+    operator fun <R, SettingType, SETTING : Setting<SettingType>> setValue(
+        thisRef: R,
+        property: KProperty<*>,
+        setting: SETTING
+    ) {
+        Settings.put(property.name, setting)
+    }
+
+    inline operator fun <R, reified SETTING> getValue(thisRef: R, property: KProperty<*>): SETTING {
```

Comment:
Not everyone is familiar with KProperty, maybe some comments with reference to external docs would be helpful